Shared/XML/CXMLNodeImpl.cpp

```diff
@@ -553,3 +553,19 @@ void CXMLNodeImpl::SetCommentText(const char* szCommentText, bool bLeadingBlankL
     SString strComment = " " + SString::Join("\n", lineList) + " ";
     pCommentNode->SetValue(strComment);
 }
+
+std::string CXMLNodeImpl::ToString()
+{
+    TiXmlDocument* pDocument = m_pDocument ? m_pDocument : static_cast<TiXmlDocument*>(m_pNode->ToDocument());
+
+    if (pDocument == nullptr)
+        return nullptr;
+
+    std::unique_ptr<TiXmlPrinter> pPrinter(new TiXmlPrinter());
```

Comment:
You don't need to heap allocate it here. Just do 
```
TiXmlPrinter printer;
```

Reply:
Fixed in #2031 - can you review/merge that PR and then I can merge master into this branch again?